includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -907,29 +905,77 @@ public function add_days_to_complete_to_lessons_query( $clauses ) {
 	}
 
 	/**
-	 * Filter the users by last activity start/end date.
+	 * Filter the courses by last activity start/end date.
 	 *
 	 * @since  4.2.0
 	 * @access private
 	 *
-	 * @param WP_User_Query $query The user query.
+	 * @param array $clauses Associative array of the clauses for the query.
+	 *
+	 * @return array Modified associative array of the clauses for the query.
 	 */
-	public function filter_users_by_last_activity( WP_User_Query $query ) {
+	public function filter_courses_by_last_activity( array $clauses ): array {
 		global $wpdb;
 
-		$start_date = DateTime::createFromFormat( 'Y-m-d', $this->get_start_date_filter_value() );
-		$end_date   = DateTime::createFromFormat( 'Y-m-d', $this->get_end_date_filter_value() );
+		$start_date = $this->get_start_date_and_time();
+		$end_date   = $this->get_end_date_and_time();
 
 		if ( ! $start_date && ! $end_date ) {
-			return;
+			return $clauses;
 		}
 
+		// Join only the last activity comment.
+		// Following the logic from `Sensei_Analysis_Overview_List_Table::get_last_activity_date()`.
+		$clauses['join'] .= " INNER JOIN {$wpdb->comments} ON {$wpdb->comments}.comment_ID = (
+			SELECT comment_ID
+			FROM {$wpdb->comments}
+			WHERE {$wpdb->comments}.comment_post_ID IN (
+				SELECT pm.post_id
```

Comment:
Yep, the problem is I couldn't find a better way around it... I did a test with 330k comments and the query takes ~1.6 seconds which is OK-ish? I couldn't make it faster when using a join or querying the lesson ids separately.

<img width="1315" alt="Screenshot on 2022-03-11 at 17-43-51" src="https://user-images.githubusercontent.com/1612178/157903773-bb6652ac-68da-420a-b188-ebaed9092f88.png">

I'm no SQL expert so please let me know if you have any idea how to improve this.

On a side note, the performance of sub-selectors is not that bad in recent years. _In some cases_, they are better than using joins. I think the Laravel ORM is using sub-selectors instead of joins exclusively.

Reply:
I did a few more tests and it turned out that using a join made the query 10% faster! https://github.com/Automattic/sensei/pull/4909/commits/43cd7ee9bfea59a8f90a0017f94191d19a5180e0